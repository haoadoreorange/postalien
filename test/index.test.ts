import { spawn } from "child_process";

describe(`Postwoman test suite`, () => {
    test(`cli output`, (done) => {
        const child = spawn(`node`, [`examples/bin.js`]);

        const chunks: Uint8Array[] = [];

        child.stdout.on(`data`, (chunk) => {
            chunks.push(chunk);
        });

        child.stdout.on(`end`, () => {
            const stdout = Buffer.concat(chunks).toString();
            expect(stdout).toMatchSnapshot();
            done();
        });

        setTimeout(() => child.kill(), 500);
    });
});
