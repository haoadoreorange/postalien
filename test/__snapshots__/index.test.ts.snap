// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postwoman test suite cli output 1`] = `
"? Which request to make ? (Use arrow keys)
  collection
  === folder1
‚ùØ     request1 
  === folder2
      request1 [15D[15C"
`;
